src/averageAges.js

```diff
@@ -37,7 +45,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWithWoman = people.filter(el => el.sex === 'f');
+  const womanWithKids = arrayWithWoman.filter(woman => {
+    return people.some(child => woman.name === child.mother);
+  });
+
+  const sumOfAges = withChildren
+    ? womanWithKids.reduce((sum, num) => sum + (num.died - num.born), 0)
+    : arrayWithWoman.reduce((sum, num) => sum + (num.died - num.born), 0);
```

Comment:
use a variable to set needed sample and word only with it. you have a lot of duplicating here